kapitan/remoteinventory/fetch.py

```diff
@@ -0,0 +1,201 @@
+       :param pool: pool object for multiprocessing
+
+       :return: None
+    """
+    temp_dir = tempfile.mkdtemp()
+
+    git_inventories = defaultdict(list)
+    http_inventories = defaultdict(list)
+
+    # To ensure no dublicate output path
+    inv_output_path = defaultdict(set)
+
+    for target_obj in target_objs:
+        try:
+            inventories = target_obj["inventories"]
+            for inv in inventories:
+                inv_type = inv["type"]
+                source_uri = inv["source"]
+                source_hash = hashlib.sha256(source_uri.encode())
+                # hashing the source and subdir together for git sources
+                # as different inventory items can have the same git uri
+                if "subdir" in inv:
+                    subdir = inv["subdir"]
+                    source_hash.update(subdir.encode())
+                if source_hash in cached.inv_sources:
+                    continue
+                # output_path is relative to inventory_path
+                # ./inventory by default
+                output_path = os.path.join(inventory_path, inv["output_path"])
+                if output_path in inv_output_path[source_uri]:
+                    continue
+                else:
+                    inv_output_path[source_uri].add(output_path)
+
+                if inv_type == "git":
+                    git_inventories[source_uri].append(inv)
+                elif inv_type in ("http", "https"):
+                    http_inventories[source_uri].append(inv)
+        except KeyError:
+            continue
+
+    git_worker = partial(fetch_git_inventories, inventory_path=inventory_path, save_dir=temp_dir)
+    http_worker = partial(fetch_http_inventories, inventory_path=inventory_path, save_dir=temp_dir)
+    [p.get() for p in pool.imap_unordered(git_worker, git_inventories.items()) if p]
+    [p.get() for p in pool.imap_unordered(http_worker, http_inventories.items()) if p]
+
+    rmtree(temp_dir)
+    logger.debug("Removed {}".format(temp_dir))
+
+
+def fetch_git_inventories(inv_mapping, inventory_path, save_dir):
+    """ Fetches remote inventories of git type as declared in target_objs
+        and recursively copies it to the output_path stored in inv_mapping
+
+        :param inv_mapping: tupple containing source and list of invetory variables
+        :param inventory_path: default or user defined inventory path
+        :param save_dir: directory to same the fetched inventory items into
+
+        :returns: None
+    """
+    source, inventories = inv_mapping
+    fetch_git_source(source, save_dir)
+    for inv in inventories:
+        repo_path = os.path.join(save_dir, os.path.basename(source))
+        repo = Repo(repo_path)
+        output_path = os.path.join(inventory_path, inv["output_path"])
+        copy_src_path = repo_path
+
+        if "ref" in inv:
+            ref = inv["ref"]
+            repo.git.checkout(ref)
+        else:
+            repo.git.checkout("master")
```

Comment:
Doublechecking here that we mention in documentation that inventory fetcher defaults to master branch?